src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java

```diff
@@ -142,9 +142,10 @@ public Screen rebuildSessionFromFrame(MenuSession menuSession, CaseSearchHelper
                 }
 
                 if (currentStep != null && currentStep != NEXT_SCREEN && entityScreen.shouldBeSkipped()) {
-                    menuSession.handleInput(screen, currentStep, needsFullInit, true, false, entityScreenContext);
-                    screen = menuSession.getNextScreen(needsFullInit, entityScreenContext);
-                    continue;
+                    if (((EntityScreen)screen).autoSelectEntities(menuSession.getSessionWrapper())) {
```

Comment:
Is there no case where we get to this line but we should not auto select but handle the input?

Reply:
I have the same question, if the conditional is false, shouldn't we do `handleInput` and then `getNextScreen`?

Reply:
Yes, that was my confusion as well here. But it seems like an entity screen without auto-select will get handled [as part of the handleInput later below](https://github.com/dimagi/formplayer/blob/persistentMenuAutoSelectFix/src/main/java/org/commcare/formplayer/services/MenuSessionFactory.java#L200)

Reply:
ah, that makes sense